dbms/programs/server/HTTPHandler.cpp

```diff
@@ -109,6 +112,9 @@ static Poco::Net::HTTPResponse::HTTPStatus exceptionCodeToHTTPStatus(int excepti
         return HTTPResponse::HTTP_BAD_REQUEST;
     else if (exception_code == ErrorCodes::SYNTAX_ERROR)
         return HTTPResponse::HTTP_BAD_REQUEST;
+    else if (exception_code == ErrorCodes::INCORRECT_DATA ||
+             exception_code == ErrorCodes::TYPE_MISMATCH)
+        return HTTPResponse::HTTP_BAD_REQUEST;
```

Comment:
Duplicate statement in different branches. Every static analyzer would argue. You can merge with the previous branch.

Reply:
Yes, but there already were a few branches returning `HTTPResponse::HTTP_BAD_REQUEST`. Because of that I actually was tempted to replace entire function body by a single switch/case statement to make it look like

```cpp
switch (exception_code) {     
    case ErrorCodes::REQUIRED_PASSWORD:     
        return HTTPResponse::HTTP_UNAUTHORIZED;     
    case ErrorCodes::CANNOT_PARSE_TEXT:     
    case ErrorCodes::CANNOT_PARSE_ESCAPE_SEQUENCE:     
    case ErrorCodes::CANNOT_PARSE_QUOTED_STRING:     
    case ErrorCodes::CANNOT_PARSE_DATE:     
    case ErrorCodes::CANNOT_PARSE_DATETIME:     
    case ErrorCodes::CANNOT_PARSE_NUMBER:     
 
    case ErrorCodes::UNKNOWN_ELEMENT_IN_AST:     
    case ErrorCodes::UNKNOWN_TYPE_OF_AST_NODE:     
    case // ...
        return HTTPResponse::HTTP_UNAUTHORIZED;
    default:
        return HTTPResponse::HTTP_INTERNAL_SERVER_ERROR;
}    
```

Reply:
Let's do it.